specifyweb/frontend/js_src/lib/components/WbImportAttachments/index.tsx

```diff
@@ -0,0 +1,220 @@
+              columns: ['Attachment'] as never,
+              data: [[]] as never,
+              specifyuser: userInformation.resource_uri,
+            }).save(),
+            attachments,
+          })
+          .then(async ({ dataSet, attachments }) => {
+            // Create SpDataSetAttachments for each attachment
+            setFileUploadProgress(true);
+            return f.all({
+              unsavedDataSetAttachments: Promise.all(
+                attachments.map(
+                  (attachment, index) =>
+                    new tables.SpDataSetAttachment.Resource({
+                      attachment: attachment as never,
+                      spdataset: dataSet.url(),
+                      ordinal: index,
+                    })
+                )
+              ),
+              dataSet,
+            });
+          })
+          .then(async ({ dataSet, unsavedDataSetAttachments }) =>
+            // Save all SpDataSetAttachments to the backend
+            f.all({
+              dataSetAttachments: ajax<
+                RA<SerializedRecord<SpDataSetAttachment>>
+              >(
+                `/api/specify/bulk/${tables.SpDataSetAttachment.name.toLowerCase()}/`,
+                {
+                  method: 'POST',
+                  headers: { Accept: 'application/json' },
+                  body: unsavedDataSetAttachments.map((dataSetAttachment) =>
+                    serializeResource(dataSetAttachment)
+                  ),
+                }
+              ).then(({ data }) =>
+                data.map((resource) =>
+                  deserializeResource(serializeResource(resource))
+                )
+              ),
+              dataSet,
+            })
+          )
+      )
+      .then(({ dataSet, dataSetAttachments }) => {
+        // Put all SpDataSetAttachments IDs into the data set
+        const data = dataSetAttachments.map((dataSetAttachment) => [
+          dataSetAttachment.id.toString(),
+        ]);
+        dataSet.set('data', data as never);
+        dataSet.set('spDataSetAttachments', dataSetAttachments);
+        return dataSet.save();
+      })
+      .then((dataSet) => navigate(`/specify/workbench/plan/${dataSet.id}/`))
+      .catch(async (error) => {
+        setFileUploadProgress(undefined);
+        setFailed();
+        raise(error);
+      });
+  };
+
+  const [dataSetName, setDataSetName] = React.useState<string>(
+    attachmentsText.attachments()
+  );
+  return isFailed ? (
+    <p>{attachmentsText.attachmentServerUnavailable()}</p>
+  ) : (
+    <>
+      {fileUploadProgress !== undefined && (
+        <Dialog
+          buttons={undefined}
+          header={attachmentsText.uploadingInline()}
+          onClose={undefined}
+        >
+          <div aria-live="polite">
+            {typeof fileUploadProgress === 'number' ? (
+              <Progress max={files.length} value={fileUploadProgress} />
+            ) : (
+              loadingBar
```

Comment:
If you go ahead with changing `fileUploadProgress` to always be a number you no longer need to look for `fileUploadProgress === true` to figure out whether it's complete and can simply check the percentage completed.

If `fileUploadProgress / files.length === 1` it would mean all files are uploaded